### PR TITLE
Added iCal feed for (upcoming) events. Needs testing

### PR DESCRIPTION
**Pull request by janwillemm**
Solves issue #51 

Was not able to test, as I could not create events due to issue #58

### CODE REVIEW

File: src/main/java/ch/wisv/events/app/controller/ICalController.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * Copyright (c) 2016  W.I.S.V. 'Christiaan Huygens'
```
Comment (TimvdLippe) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: Shouldnt this be at the top of the file? I think this is now the javadoc of this class

File: src/main/java/ch/wisv/events/utils/ICalendarBuilder.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+public class ICalendarBuilder {
+
+    /**
+     * Generates a iCal object for a list of events
+     * @param events
+     * @return
+     */
+    public static ICalendar createIcalEventList(List<Event> events){
+        ICalendar iCal = new ICalendar();
+        events.forEach(event -> {iCal.addEvent(createICalEvent(event));});
```
Comment (TimvdLippe) [inline src/main/java/ch/wisv/events/utils/ICalendarBuilder.java]: `{` and `}` are unnecessary I think

File: src/main/java/ch/wisv/events/app/controller/ICalController.java (hunk 1 above)
Comment (janwillemm) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: I copied it from some other files and mimicked the placement. But I do agree with you.

File: src/main/java/ch/wisv/events/app/controller/ICalController.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+     * Then we copy the contentstream of the file to the output stream of the response.
+     * @param ical
+     * @param response
+     * @throws IOException
+     */
+    private void presentIcalFile(ICalendar ical, HttpServletResponse response){
+        try{
+            // Create a temporary file to for the iCalWriter to write to.
+            File file = File.createTempFile("ical", ".ics");
+            ICalWriter writer = new ICalWriter(file, ICalVersion.V2_0);
```
Comment (praseodym) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: You can also write to an `OutputStream` (e.g. `ByteArrayOutputStream`), so maybe use that instead.

File: src/main/java/ch/wisv/events/app/controller/ICalController.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            writer.write(ical);
+            writer.close();
+
+            // Make the outputstream the same as our temporary file
+            IOUtils.copy(new FileInputStream(file), response.getOutputStream());
+
+            // Attaches the file to the response
+            response.flushBuffer();
+        } catch (IOException e) {
+            throw new RuntimeException("IOError writing ICalendar to response output stream");
```
Comment (praseodym) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: -> `throw new RuntimeException("...", e);`

File: src/main/java/ch/wisv/events/app/controller/ICalController.java (hunk 3 above)
Comment (janwillemm) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: Not sure how I missed that, changed it accordingly, thanks

File: src/main/java/ch/wisv/events/utils/ICalendarBuilder.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+     * Generates a iCal object for a list of events
+     * @param events
+     * @return
+     */
+    public static ICalendar createIcalEventList(List<Event> events){
+        ICalendar iCal = new ICalendar();
+        events.forEach(event -> iCal.addEvent(createICalEvent(event)));
+
+        iCal.addName("CH Event Calendar");
+        iCal.setColor("turquoise");
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/utils/ICalendarBuilder.java]: Let's not start with all the self hypes and just use the one of the default CH colours.

File: src/main/java/ch/wisv/events/app/controller/ICalController.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    /**
+     * Default constructor
+     */
+    public ICalController(EventService eventService){this.eventService = eventService;}
+
+    /**
+     * Get request on "/events/iCal" will present the ical with the events
+     *
+     */
+    @RequestMapping(value = "/iCal", method = RequestMethod.GET, produces = "text/calendar; charset=utf-8")
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: Can be replaced by `@GetMapping()`

File: src/main/java/ch/wisv/events/app/controller/ICalController.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        // Getting the iCal with the current available events
+        ICalendar ical = ICalendarBuilder.createIcalEventList(eventService.getAvailableEvents());
+        presentIcalFile(ical, response);
+    }
+
+    /**
+     * Get request on /iCal/upcoming will present the ical with the upcoming events
+     * @param response
+     */
+    @RequestMapping(value = "/iCal/upcoming", method = RequestMethod.GET, produces = "text/calendar; charset=utf-8")
```
Comment (svenpopping) [inline src/main/java/ch/wisv/events/app/controller/ICalController.java]: Can be replaced by `@GetMapping()`

File: src/main/java/ch/wisv/events/utils/ICalendarBuilder.java (hunk 5 above)
Comment (janwillemm) [inline src/main/java/ch/wisv/events/utils/ICalendarBuilder.java]: I've removed the colouring, better let the user choose it themselves

**Pull request merged**